### ISSUE TITLE
Duplicate Entries in S3 Listings with Redis Filer Enabled

### ISSUE THREAD

**Issue by sample** [seaweedfs/seaweedfs#5426]
### Bug description

When using the Redis filer (and potentially other filers lacking ListDirectoryPrefixedEntries implementation), executing s3cmd put followed by s3cmd ls on this file results in duplicate entries for the uploaded file. Specifically, after uploading a file using `s3cmd put test1 s3://test-bucket`, running `s3cmd ls s3://test-bucket/test1` outputs the same file entry 10,000 times.

#### Commands used to start services:
```./weed/weed -config_dir=<path_to>/config -v 8 -logtostderr server -dir=<path_to>/data -s3 -filer -volume.max=0 -master.volumeSizeLimitMB=100```

#### OS version:
the bug is reproduced regardless of the operating system

#### Output of weed version:
master branch last commit  `36fec34c472266fe2a8f087af7de3945dc198726`
```./weed version: version 30GB 3.64  darwin arm64```

#### Contents of filer.toml:
```[filer.options]
recursive_delete = true
[redis2]
enabled = true
address = "localhost:6379"
password = ""
database = 0
superLargeDirectories = []
```

#### Expected behavior:
After uploading a file with s3cmd put, running s3cmd ls on the bucket or directory where the file was uploaded should list each file exactly once, without duplicates.

### Additional context
The issue appears to be related to the `(fsw *FilerStoreWrapper) prefixFilterEntries` function in `filerstore_wrapper.go`.

**Comment by kmlebedev** [comment]
Hi, please check on the [dev](https://github.com/seaweedfs/seaweedfs/releases/tag/dev) version there was recently a listing fix for nosql

**Comment by sample** [comment]
All tests passed ```================================================================================================ 125 passed in 52.32s ================================================================================================
______________________________________________________________________________________________________ summary _______________________________________________________________________________________________________
  py: commands succeeded
  congratulations :)```

**Comment by kmlebedev** [comment]
Another option is to try rolling back the PR locally:
https://github.com/seaweedfs/seaweedfs/pull/5350 and https://github.com/seaweedfs/seaweedfs/pull/5411

**Comment by sample** [comment]
i tried this code:
```diff --git a/weed/filer/filerstore_wrapper.go b/weed/filer/filerstore_wrapper.go
index cc3e58363..f8a8831d3 100644
--- a/weed/filer/filerstore_wrapper.go
+++ b/weed/filer/filerstore_wrapper.go
@@ -304,6 +304,9 @@ func (fsw *FilerStoreWrapper) prefixFilterEntries(ctx context.Context, dirPath u
                        }
                }
                if count < limit && lastFileName <= prefix {
+                       if lastFileName == prefix {
+                               break
+                       }
                        notPrefixed = notPrefixed[:0]
                        lastFileName, err = actualStore.ListDirectoryEntries(ctx, dirPath, lastFileName, false, limit, func(entry *Entry) bool {
                                notPrefixed = append(notPrefixed, entry)```

this fixes the problem, the tests pass as well. But I don't understand if it can still cause any problems

**Comment by kmlebedev** [comment]
> this fixes the problem, the tests pass as well. But I don't understand if it can still cause any problems

Cool, yes, it seems that only s3 test is not enough. There was also an idea to set up testing of s3 via rclone.
```
1. make server
2. git clone rclone
3. configure rclone on seawedfs
4. cd rclone/backend/s3
5. go test -v -remote seaweedfs:rclonetest  
```

**Comment by sample** [comment]
`go test -v -remote seaweedfs:rclonetest` generates a large number of errors, are there any specific tests that need to be run?

**Comment by kmlebedev** [comment]
> `go test -v -remote seaweedfs:rclonetest` generates a large number of errors, are there any specific tests that need to be run?

This is the point, I haven’t figured out how to run a specific set of tests that I previously passed :(

**Comment by sample** [comment]
are there any steps I could take to solve this?

**Comment by kmlebedev** [comment]
> are there any steps I could take to solve this?

I'm waiting for @chrislusf  to merge https://github.com/seaweedfs/seaweedfs/pull/5427 into the master

Then you can make a PR with your fix.

**Comment by kmlebedev** [comment]
@sample Hello, let's go PR let's see how the s3 tests go.

**Comment by sample** [comment]
fixed in #5440
